workflows/plugins/include/utils/cds.py

```diff
@@ -0,0 +1,122 @@
+import logging
+
+from idutils import is_arxiv
+from inspire_utils.record import get_value
+
+logger = logging.getLogger(__name__)
+
+LITERATURE_PID_TYPE = "literature"
+
+
+def has_any_id(cds_record):
+    cds_id = cds_record.get("id") or get_value(
+        cds_record, "metadata.control_number", []
+    )
+    if not cds_id:
+        logger.info(f"Cannot extract CDS id from CDS response: {cds_record}")
+        return False
+    return any(
+        [
+            get_value(cds_record, "metadata.other_ids", []),
+            get_value(cds_record, "metadata.eprints", []),
+            get_value(cds_record, "metadata.dois.value", []),
+            get_value(cds_record, "metadata.report_numbers.value", []),
+        ]
+    )
+
+
+def search_and_return_single(
+    hook,
+    query_url,
+    log_message,
+):
+    """
+    Build query params from the URL, call search_records,
+    and if exactly one hit is found
+    return its metadata with a log; otherwise return None.
+    """
+    query_params = {
+        "q": query_url,
+        "fields": "control_number",
+    }
+    logger.info(
+        f"Searching for record with query: {query_params}",
+    )
+    resp = hook.search_records(pid_type=LITERATURE_PID_TYPE, query_params=query_params)
+    logger.info(
+        f"Response from search: {resp}",
+    )
+    hits = resp.get("hits", {}).get("hits", [])
+    logger.info(
+        f"Found {len(hits)} hits for query",
+    )
+    if len(hits) == 1:
+        record = hits[0]["metadata"]
+        control_number = record.get("control_number")
+        logger.info(f"{log_message} Control number: {control_number}")
+        return control_number
+    return None
```

Comment:
Just to make the code a bit more readable (less nested), I would re-write it with early return

```suggestion
    if not len(hits) == 1:
        return None
    record = hits[0]["metadata"]
    control_number = record.get("control_number")
    logger.info(f"{log_message} Control number: {control_number}")
    return control_number

```